Providers/Profile.swift

```diff
@@ -1082,33 +1099,54 @@ public class BrowserProfile: Profile {
         }
 
         private func syncClientsWithDelegate(delegate: SyncDelegate, prefs: Prefs, ready: Ready) -> SyncResult {
-            log.debug("Syncing clients to storage.")
-            let clientSynchronizer = ready.synchronizer(ClientsSynchronizer.self, delegate: delegate, prefs: prefs)
-            return clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info)
+            return statsWrapEngineSync("clients") { statsDelegate in
```

Comment:
Why not have the `SyncDelegate` also be the stats delegate, and have the stats API encapsulate the engine part? So these functions would all stay the same, then elsewhere, at point of use, you'd "check out" a stats recorder for the implementing engine from the delegate.